PhotonSieveDevelopment.js

```diff
@@ -24,92 +24,96 @@ class Sieve {
         this.totalPinholes = totalPinholes
         this.totalArea = totalArea
         this.fNumber = fNumber
+        this.rings = []
+    }
+
+    addRing(ring) {
+        this.rings.push(ring);
+        this.totalArea = this.totalArea + ring.openArea;
+        this.totalPinholes = this.totalPinholes + ring.pinholes.length;
+        let sieveEquivalentAperture = Math.sqrt(this.totalArea / Math.PI) * 2;
+        this.fNumber = FD / sieveEquivalentAperture;
     }
 }
 
 class Ring {
-    constructor(zone, centerlineRadius, ringWidth, numOfPinholes, openArea){
+    constructor(zone, centerlineRadius){
         this.zone = zone
         this.centerlineRadius = centerlineRadius
-        this.ringWidth = ringWidth
-        this.numOfPinholes = numOfPinholes
-        this.openArea = openArea
+        this.openArea = 0
+        this.pinholes = []
     } 
+
+    addPinhole(pinhole) {

```

Comment:
In theory we should call getConflicts here and throw an error, but having main code do conflict check is prob fine. Reason I say may be better here is technically this method allows adding invalid pinholes